packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -368,6 +416,44 @@ export async function activate(context: vscode.ExtensionContext) {
   telemetryService.initializeService(context, machineId);
   telemetryService.showTelemetryMessage();
 
+  // Task View
+  const treeDataProvider = vscode.window.registerTreeDataProvider(
+    'sfdx.force.tasks.view',
+    taskViewService
+  );
+  context.subscriptions.push(treeDataProvider);
+
+  // Set internal dev context
+  const internalDev = sfdxCoreSettings.getInternalDev();
+
+  vscode.commands.executeCommand(
+    'setContext',
+    'sfdx:internal_dev',
+    internalDev
+  );
+
+  if (internalDev) {
```

Comment:
Doing this allows for setting a limited number of functionality in the core extension when the user has set internal development flag as true. E.g. no org picker or command palette commands.